### PR TITLE
test: add some unit tests

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What this PR does / why we need it**:
Add a few unit tests for Azure reconciler code and clean up some unused code.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: controllers/azuremachine_reconciler_test.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+	"sigs.k8s.io/cluster-api-provider-azure/cloud/scope"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha2"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+func TestGetControlPlaneMachines(t *testing.T) {
+	controlPlaneMachine := clusterv1.Machine{
+		ObjectMeta: v1.ObjectMeta{
+			Name:   "control-plane-0",
+			Labels: map[string]string{},
```
Comment (awesomenix) [inline controllers/azuremachine_reconciler_test.go]: You could add the label right here
```golang
Labels: {clusterv1.MachineControlPlaneLabelName: "cp-0"},
```

File: controllers/azuremachine_reconciler_test.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+	controlPlaneMachine.ObjectMeta.Labels[clusterv1.MachineControlPlaneLabelName] = "cp-0"
+
+	agentMachine := clusterv1.Machine{
+		ObjectMeta: v1.ObjectMeta{
+			Name:   "machine-0",
+			Labels: map[string]string{},
+		},
+	}
+
+	cases := []struct {
```
Comment (awesomenix) [inline controllers/azuremachine_reconciler_test.go]: could you please add a test case for multiple control plane machines as well

File: controllers/azuremachine_reconciler_test.go [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+	}
+
+	for _, l := range azure.SupportedAvailabilityZoneLocations {
+		s.machineScope.AzureCluster.Spec.Location = l
+		if s.isAvailabilityZoneSupported() != true {
+			t.Errorf("isAvailabilityZoneSupported should return true for supported region %s but returned %t", l, s.isAvailabilityZoneSupported())
+		}
+	}
+
+	unSupportedLocations := []string{
```
Comment (awesomenix) [inline controllers/azuremachine_reconciler_test.go]: Seems like an annoying test case, once somebody updates the list of supported, they need to update this unsupported set as well.

Recommend testing some random location name than a valid one. Example "randomregion" or something like that.

File: controllers/azuremachine_reconciler_test.go (hunk 2 above)
Comment (CecileRobertMichon) [reply controllers/azuremachine_reconciler_test.go]: done

File: controllers/azuremachine_reconciler_test.go (hunk 3 above)
Comment (CecileRobertMichon) [reply controllers/azuremachine_reconciler_test.go]: yep good point. Changed.

**Comment by CecileRobertMichon** [comment]
related to #273

**Pull request merged**